app/pages/cif/project-revision/[projectRevision]/index.tsx

```diff
@@ -89,14 +91,32 @@ export function ProjectRevision({
     });
   };
 
-  if (!query.projectRevision.id) return null;
+  const triggerFormSelfValidation = (formObject: any): [] => {
+    formObject.onSubmit({
+      preventDefault: () => {},
+      persist: () => {},
+    });
 
-  const onFormErrors = (formId: string, incomingErrors: Array<any>) => {
-    setErrors({ ...errors, [formId]: incomingErrors });
+    // Effectively validating the form a second time to retrieve the errors
+    const validationResult = formObject.validate(formObject.state.formData);
```

Comment:
Unfortunately rjsf's [onSubmit](https://github.com/rjsf-team/react-jsonschema-form/blob/8335ad1966c7e249225d9dce50766750785d8e77/packages/core/src/components/Form.js#L307) (which we need to trigger internal error display) doesn't return a set of errors, so we need to re-validate to grab them

Reply:
I wish there was a better way to do this, other than using refs...
The only thing I can think of is having a single form, and having an adapter layer to merge the various form_change into a single form data, but that comes with its own complexity, and may not be better

Reply:
I was looking at this...rjsf has methods to merge schemas and forms, but that would be a large amount of work for something that we are likely to refactor at a later point